full pass on Style Guide

taroface opened the pull request:
- Reorganized and consolidated Style Guide headings
- Added new section on "Prescriptive writing"
- Attempted to make Style Guide conform to its own guidelines (this work will be ongoing):
  - Writing is more concise, prescriptive, and 'active'
  - Guidance is formatted in a generally consistent manner
  - Correct punctuation is used, etc.
- Added new guidance on "in" vs. "on" word usage per this [earlier exchange](https://cockroachlabs.slack.com/archives/C01NFCXULTX/p1729193283515209)
- Corrected the guidance on "Cockroach Labs's" (this is **not** allowed)
- Added some guidance on how & when to make diagrams
- Added new section on avoiding "Latinisms"
- Added guidance on how code blocks should be presented
- Added some new guidance opportunistically (on straight quotes, colons over dashes, ...)

jhlodin commented on StyleGuide.md:
```diff
@@ -5,98 +5,219 @@ The following guidance is provided to benefit CockroachDB docs authors and revie
 
-- Avoid using "please" when giving an instruction, except when asking the user to go outside the scope of the task (such as contacting Cockroach Labs or filing a support issue).
+- **Avoid:** Define additional settings as desired.  
+  **Prefer:** You can define additional settings in the configuration file. To change the default host and port, set `host` and `port` respectively. To change the path, ...
```
This implies that we should spell out every configuration option available, which can get unwieldy fast. Especially in deployment guides where we may want to offer a "one size fits all" initial configuration but leave room for optional configurations. In those circumstances I would rather link out if there's a good docs target or even leave out the sentences after the first generic one.

jhlodin replied:
https://www.cockroachlabs.com/docs/stable/deploy-cockroachdb-with-cockroachdb-operator#initialize-the-cluster Step 6 here is a hand-wavey note that essentially says "if you know you need to make other changes to your configuration yaml, now is a good time to make those" without going into details of every option. I believe it's compliant with the first sentence of your **Prefer** example, but it refers to only a couple example configs and doesn't have task-oriented sentences like the "To change the ... set ..." in that example.

I would just want to clarify in the style guide if this approach is acceptable when there are too many specific configs to spell out in that single instruction

taroface replied:
I see. The style guidance here is specifically intended to guardrail us against hand-waving. It's of course not possible to mention or prescribe a value for every configurable detail of every customer's deployment. But from past/ongoing experiences, phrases like "If you know you need ..." or "Modify other relevant parts of the configuration ..." will prompt at least some customers to respond, "How do I know I need to?" and "Which parts are relevant?". A subtle balancing act might be to simply state in a callout that "Depending on your configuration, you may need to set additional parameters in the XYZ fields". 

@Flat5flat9 wondering if you have any thoughts here.

jhlodin commented on StyleGuide.md:
```diff
@@ -5,98 +5,219 @@ The following guidance is provided to benefit CockroachDB docs authors and revie
+- Databases on the cluster
+- Jobs running on the cluster
+
+## Inclusivity
```
Suggest a quick subsection under here to avoid idioms and terminology that could be confusing when auto-translated (which should be assumed given that we don't have a localization team).

Suggest something like:

```
### Avoid idioms and other difficult-to-translate terminology

The majority of our non English-speaking readers use a browser translation tool to read our documentation in a preferred language. Avoid using idioms or non industry-standard terms that these tools may translate literally.

- **Avoid:** Configure roles and permissions to draw a line between allowed and disallowed actions.
  **Prefer:** Configure roles and permissions to allow and disallow actions.
```

jhlodin commented on StyleGuide.md:
```diff
@@ -206,124 +322,142 @@ Try to address as many of the suggestions as possible. If Vale flags a word that
+Code samples often include placeholder values, to be replaced by values specific to a user's environment. To denote that a value in a code sample is a placeholder value that should be replaced, use curly brackets (`{}`).
 
-Reference issues and pull requests by their corresponding number, prepended with `#`.
+- **Example:** `SELECT * FROM TABLE {mytable};`.
```
Suggest also adding an example from the API docs where these are commonly used in URLs, like

```
- **Example:** `--url https://cockroachlabs.cloud/api/v1/clusters/{cluster_id}/nodes`
```

jhlodin commented on StyleGuide.md:
```diff
@@ -206,124 +322,142 @@ Try to address as many of the suggestions as possible. If Vale flags a word that
+**Best practices:**
 
-To insert a note, use the following code:
+- Avoid placing callouts adjacent to each other.
```
I would add "Avoid placing a callout immediately below a header without introductory text for the section" but IIRC we've done that in a couple places.

jhlodin commented on StyleGuide.md:
```diff
@@ -206,124 +322,142 @@ Try to address as many of the suggestions as possible. If Vale flags a word that
+
+Use images to clarify a topic, but only use them as needed. Images are either:
+
+- **Screenshots:** Depict a UI element. Screenshots should show enough of the UI that the user can easily orient themselves and understand what they are being shown. If a screenshot needs an annotation, use a red box.
```
```suggestion
- **Screenshots:** Depict a UI element. Screenshots should only show enough of the UI that the user can easily orient themselves and understand what they are being shown. If a screenshot needs an annotation, use a red box.
```

Would like to emphasize not including page elements that are not relevant to the screenshot's purpose.

jhlodin commented on StyleGuide.md:
```diff
@@ -206,124 +322,142 @@ Try to address as many of the suggestions as possible. If Vale flags a word that
+  **Note:** Screenshots are difficult to keep current, and impact accessibility. Use a screenshot only if it is necessary for the user to accomplish a task or understand a feature. For example, in a page on troubleshooting cluster performance, a screenshot of a Contention metric graph in the DB Console might be used to depict a cluster with high contention. The same screenshot would not be necess ...
 
-Use a warning to express that a piece of information is critical to understand to prevent data loss, security vulnerabilities, or unexpected behavior.
+  To optimize accessibility and SEO, a screenshot should be accompanied by text and [alt-text](#write-accessible-documentation) describing its contents.
```
Could we include an example here showing the correct syntax to do this? Like:

```<img src="{{ 'images/v24.2/topology-patterns/topology_multi-region_hardware.png' | relative_url }}" alt="Multi-region hardware setup" style="max-width:100%" />```

jhlodin commented on StyleGuide.md:
```diff
@@ -206,124 +322,142 @@ Try to address as many of the suggestions as possible. If Vale flags a word that
-<warning text goes here>
-{{site.data.alerts.end}}
-~~~
+  - [Monodraw](https://monodraw.helftone.com/)
+  - [Omnigraffle](https://www.omnigroup.com/omnigraffle)
```
Suggest appending the use cases for each. IMO these are
- Monodraw (useful for abstract concept diagrams)
- Omnigraffle (useful for diagrams with branded elements like logos and colors)

jhlodin commented on StyleGuide.md:
```diff
@@ -206,124 +322,142 @@ Try to address as many of the suggestions as possible. If Vale flags a word that
+  - [Omnigraffle](https://www.omnigroup.com/omnigraffle)
 
-#### CockroachDB version callout
+  Icons for diagrams can be sourced from: https://github.com/cockroachlabs/cockroach-studios/tree/main/icons
```
```suggestion
  Icons and colors for diagrams can be sourced from: https://github.com/cockroachlabs/cockroach-studios/tree/main/icons
```

jhlodin replied:
The `.gstencil` file in that directory is an Omnigraffle template listing our branded primary/secondary/extended color palette.

rmloveland commented on StyleGuide.md:
```diff
@@ -5,98 +5,219 @@ The following guidance is provided to benefit CockroachDB docs authors and revie
+
+### Third-party products
+
+In general, align third-party branding with that brand's usage. Do not use the shortened versions of product names.
```
i have received feedback in the past to avoid linking to 3rd party products' docs unless absolutely necessary for clarity.  IDK if that is appropriate for inclusion in this section, or if we would like to discuss it more separately. I don't have a strong opinion _except_ that such external links often break because humans break their websites constantly

jhlodin replied:
I have the possibly total opposite opinion, which is that I don't want us to be responsible for documenting third party product flows and would rather link out to their documentation wherever possible for those. Easier to identify and fix a broken link than correct text on our site that describes an outdated third-party procedure.

taroface replied:
Linking out to external products is absolutely necessary for things like migration docs. I think I can add some follow-up guidance on external links that is consistent with both of your comments; stay tuned.

jhlodin reviewed (commented):
Broadly looks good to me, left a couple comments but no major disagreements from me. Will add comments later if other suggestions come to mind.

rmloveland commented on StyleGuide.md:
```diff
@@ -206,124 +322,142 @@ Try to address as many of the suggestions as possible. If Vale flags a word that
-The first occurrence of a product name within a docs page should use its full name. At the writer's discretion, subsequent occurrences may be shortened to "Basic", "Advanced", or "Cloud", unless a writer (or reviewer) senses contextual ambiguity that could be improved by using the full product name. In long pages, it may be helpful to use the full name for each occurrence in a new sentence or if  ...
+    The standby cluster will show the `main` virtual cluster is in a `replicating` state.
 
-## Page components
```
i think this was a separate PR but i am kinda bummed that we removed the stuff about consistent page structure, e.g. 

- before you begin
- ...
- see also

since i think there is a lot of value of having ~all the pages in the docs share a rough structure for easier scanning, since folks aren't usually "reading" carefully but "scanning"

however maybe this doesn't matter in agent world (tm)

taroface replied:
I don't recall seeing or removing those sections. I see that some TOC links corresponding to those were removed in https://github.com/cockroachdb/docs/pull/20170, but I think that content was previously removed (not sure if by me). I'll dig around for it in the git history.

rmloveland replied:
looks like it was removed in this PR: https://github.com/cockroachdb/docs/pull/19334/files

rmloveland reviewed (approved):
LGTM!!! thanks for going through this and doing these thoughtful updates!

PS  All my comments below are things that can be revisited in other conversations, they're not really about this PR, apologies for the digressions

taroface commented on StyleGuide.md:
```diff
@@ -144,59 +264,55 @@ Terminology around "kill" vs. "stop" vs. "terminate" is nuanced, as described [i
-- Avoid using _and/or_ unless space is very limited (e.g., in a table). Instead, decide whether **and** or **or** can stand alone or make use of **both** when the inclusivity must be explicit, e.g., **x or y or both**.
+- Use the [Oxford (aka serial) comma](https://en.wikipedia.org/wiki/Serial_comma).
+- Use straight `'` rather than curved `‘` quotes.
+- Limit semicolon usage. Instead, use two simple sentences.
```
@rmloveland This was directed specifically at **you** (jk it was already there)

taroface commented:
Finally addressed review comments, replaced some previously deleted (from the old style guide and CONTRIBUTING.md) content on page sections, and cross-linked from Style to Markdown guides, if anyone would like to re-review!
